src/Graph.ts

```diff
@@ -349,16 +345,45 @@ export default class Graph {
 
 				this.link();
 
+				const entryModules = entryAndChunkModules.slice(0, entryModuleIds.length);
+
+				let manualChunkModules: { [chunkName: string]: Module[] };
+				if (manualChunks) {
+					manualChunkModules = {};
+					for (let chunkName of Object.keys(manualChunks)) {
+						const chunk = manualChunks[chunkName];
+						manualChunkModules[chunkName] = chunk.map(entryId => {
+							const entryIndex = entryAndManualChunkIds.indexOf(entryId);
+							return entryAndChunkModules[entryIndex];
+						});
+					}
+				}
+
 				const { orderedModules, dynamicImports, dynamicImportAliases } = this.analyseExecution(
 					entryModules,
-					!preserveModules
+					!preserveModules,
+					manualChunkModules
 				);
 
+				if (entryModuleAliases) {
```

Comment:
You've brought up an interesting case, and this is really good to poke holes in the model. The difficulty with supporting this case is there are a few questions in relating it to the aliasing model:

* Should the chunk alias for `main1` be called `main1` or `initial`?
* Should the chunk name for `main` use the `chunkNames` rule or the `entryNames` rule? Since it is defined as both a manual chunk and an entry, we don't currently have a way to deal with this.

Note from a chunking point of view (naming aside), this scenario is equivalent to just having:

```js
{
  input: ['main2'],
  manualChunks:{
    initial: ['main1']
  },
  ...
}
```

One alternative might be to have the manual chunk called `initial` get created, and then have the input become simply a reexporting facade module pointing to the manual chunk? That might be a way around the naming model issues here.

Reply:
In fact the above would provide a general solution to the duplicate entry error, allowing us to have a way of always dealing with this case, removing the error entirely.

Reply:
To summarize the approach taken in the latest commit:

> Should the chunk alias for main1 be called main1 or initial?

The manual chunk is created, and then seen to be an entry module facade - therefore it takes the place of `main1`, and gets called `main1`. If the chunk had had other modules in it then a new reexporting facade would have been used to the chunk. 

> Should the chunk name for main use the chunkNames rule or the entryNames rule? Since it is defined as both a manual chunk and an entry, we don't currently have a way to deal with this.

When a chunk corresponds to an entry point facacde as in this case, the manual chunk name does not apply at all, with the entry naming scheme precedence.

Reply:
This will still require two separate downloads but I guess this is certainly the cleanest approach 👍